src/azure-cli/azure/cli/command_modules/cdn/custom.py

```diff
@@ -94,6 +100,60 @@ def update_endpoint(instance,
     return params
 
 
+def show_endpoint_waf_policy_link(client: EndpointsOperations,
+                                  resource_group_name: str,
+                                  profile_name: str,
+                                  endpoint_name: str):
+
+    from azure.mgmt.cdn.models import (EndpointPropertiesUpdateParametersWebApplicationFirewallPolicyLink)
+
+    link = client.get(resource_group_name, profile_name, endpoint_name).web_application_firewall_policy_link
+    if link is not None:
+        return link
+    return EndpointPropertiesUpdateParametersWebApplicationFirewallPolicyLink(id=None)
```

Comment:
Can you help understand this part?
When link resource cannot be got, we will get Error, right? And in this way, we should just throw this error and with `custom_show_command` build in exception handler, cli will exit with exit code 3 and error message.

Why do you want to return such object?

Reply:
If the endpoint exists but there is no link, `client.get(resource_group_name, profile_name, endpoint_name).web_application_firewall_policy_link` will simply return `None`, which `custom_show_command` does not handle. I've changed it to raise a CLIError instead of an empty WAF policy link model object in this case, which is the practice used by the cli core util function `find_child_item`.